internal/upgrader/upgrader_test.go

```diff
@@ -120,12 +132,12 @@ var _ = Describe("Upgrade", func() {
 
 	When("performing a dry run", func() {
 		It("should print out service GUIDs and not attempt to upgrade", func() {
-			var lines []string
-			fakeLog.PrintfStub = func(format string, a ...any) {
-				lines = append(lines, fmt.Sprintf(format, a...))
-			}
+			logr := logger.New(time.Minute)
```

Comment:
note: Can we next time do this kind of improvements in a separate PR? Just so it is easier to review.

Reply:
Force-pushed a new PR history. (The 3 commits are all different)

As we have discussed in via Zoom, this isn't a aesthetic improvement.
`fakeLog.PrintfStub` prevented some new code from being tested correctly.

This issue is addressed in the new commit history.